services/api/service.go

```diff
@@ -1595,6 +1604,185 @@ func (api *RelayAPI) checkBuilderEntry(w http.ResponseWriter, log *logrus.Entry,
 	return builderEntry, true
 }
 
+// Checks the quality of the TOB txs, if it is the txs expected in a TOB
+func (api *RelayAPI) checkTobTxsStateInterference(txs []*types.Transaction) error {
+	if len(txs) > 2 {
+		return fmt.Errorf("we support only 1 tx on the TOB currently, got %d", len(txs))
+	}
+
+	firstTx := txs[0]
+	if firstTx.To() == nil {
+		return fmt.Errorf("contract creation cannot be a TOB tx")
+	}
+	// This address is from the kurtosis local devnet. Need to expand state interference checks
+	if *firstTx.To() != common2.HexToAddress("0xB9D7a3554F221B34f49d7d3C61375E603aFb699e") {
+		return fmt.Errorf("TOB tx can only be sent to uniswap v2 router")
+	}
+
+	// TODO - add check for whether the tx is a eth/usdc swap
+	return nil
+}
+
+// This method first checks whether the payouts are valid, then checks whether the txs are valid w.r.t state interference
+func (api *RelayAPI) checkTxAndSenderValidity(txs []*types.Transaction) error {
+	// TODO - Payouts still need to be modelled
+	// TODO - expand state interference checks as in checkTobTxsStateInterference using call tracing
+
+	if len(txs) == 0 {
+		return fmt.Errorf("Empty TOB tx request sent!")
+	}
+	if len(txs) == 1 {
+		return fmt.Errorf("We require a payment tx along with the TOB txs!")
+	}
+
+	// Start: Payout checks
+	lastTx := txs[len(txs)-1]
+
+	if lastTx.To() != nil && *lastTx.To() != api.relayerPayoutAddress {
+		return fmt.Errorf("We require a payment tx to the relayer along with the TOB txs!")
```

Comment:
ToB especially, i dont understand why it pays the relay. if its ToB then it can just pay the proposer

Reply:
i have done exactly this in https://github.com/bharath-123/pepc-boost-relay/pull/19. It is much easier and fits payouts better.